### PR TITLE
feat: Added Batch account module

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

- Added Batch account module

| Pipeline |
| - |
| [avm.res.batch.batch-account](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.batch.batch-account.yml) |

### CODE REVIEW

File: avm/res/batch/batch-account/tests/e2e/waf-aligned/main.test.bicep
Comment (eriqua) [inline avm/res/batch/batch-account/tests/e2e/waf-aligned/main.test.bicep:4]: Ah this typo! Any automation to be updated?

```suggestion
metadata description = 'This instance deploys the module in alignment with the best-practices of the Azure Well-Architected Framework.'
```

File: avm/utilities/pipelines/sharedScripts/Set-ModuleReadMe.ps1
Comment (eriqua) [inline avm/utilities/pipelines/sharedScripts/Set-ModuleReadMe.ps1:1110]: 👻

File: .github/workflows/avm.res.batch.batch-account.yml
Comment (eriqua) [inline .github/workflows/avm.res.batch.batch-account.yml:1]: Wait for #563

File: avm/utilities/pipelines/sharedScripts/Set-ModuleReadMe.ps1
Comment (AlexanderSehr) [reply avm/utilities/pipelines/sharedScripts/Set-ModuleReadMe.ps1:1110]: Not a ghost @eriqua. If we don't have this change in (before #546) is merged, readme's would become outdated again.

File: avm/res/batch/batch-account/tests/e2e/waf-aligned/main.test.bicep
Comment (AlexanderSehr) [reply avm/res/batch/batch-account/tests/e2e/waf-aligned/main.test.bicep:4]: No, it's actually manual

File: avm/utilities/pipelines/sharedScripts/Set-ModuleReadMe.ps1
Comment (AlexanderSehr) [reply avm/utilities/pipelines/sharedScripts/Set-ModuleReadMe.ps1]: gone it is

File: avm/res/batch/batch-account/main.bicep
Comment (eriqua) [inline avm/res/batch/batch-account/main.bicep:85]:
```suggestion

var identity = !empty(managedIdentities) ? {
```

File: avm/res/batch/batch-account/main.bicep
Comment (eriqua) [inline avm/res/batch/batch-account/main.bicep:324]:
```suggestion
```

File: avm/res/batch/batch-account/main.bicep
Comment (AlexanderSehr) [reply avm/res/batch/batch-account/main.bicep:85]: If wonder if that's worth commenting as it requires me to re-render the template. But regardless - I'd be against adding an empty line here as I clustered them together on purpose (i.e., they belong together)

Comment (eriqua) [reply avm/res/batch/batch-account/main.bicep:85]: Just thinking it would improve readability to separate each var. I won't block the merge on such a small detail though.

File: avm/res/batch/batch-account/main.bicep
Comment (eriqua) [inline avm/res/batch/batch-account/main.bicep:50]: What about a `networkProfile` object hosting this and the next parameter?

Comment (AlexanderSehr) [reply avm/res/batch/batch-account/main.bicep:50]: In that case we'd need another User Defined Type. Not saying we shouldn't, but this would need to be updated in all modules with PE (+ specs) and I suggest to do this  in a separate issue + PR.

File: avm/res/batch/batch-account/main.bicep
Comment (AlexanderSehr) [reply avm/res/batch/batch-account/main.bicep:85]: I'd appreciate it if we don't 🙏

File: avm/res/batch/batch-account/main.bicep
Comment (eriqua) [reply avm/res/batch/batch-account/main.bicep:50]: Do we have a spec saying objects cannot be used unless typed with udt? I agree that would be the end goal but for the moment we made exceptions, e.g. for child resources, and not only. Other modules have networkAcls in an object, without udt (e.g. cognitive services). NetworkAcls and networkProfile are basically the same thing. So yes, let's agree on something.

Comment (AlexanderSehr) [reply avm/res/batch/batch-account/main.bicep:50]: For the time being, I would only like to enforce UDT for what we prepared with extension resources. If somebody wants to go the extra mile and add any UDT on top of that, be my guest (e.g., @ChrisSidebotham). But blocking PRs for that seems wrong to me as we have more than enough work porting the CARML modules as is, yet alone optimizing them for UDTs.

Comment (eriqua) [reply avm/res/batch/batch-account/main.bicep:50]: I'm not asking for UDT, I'm suggesting to use an object for networkProfile, to align to what done for networkAcls for other modules. Adding udt next will be easier if we already have an object.

Comment (AlexanderSehr) [reply avm/res/batch/batch-account/main.bicep:50]: I see. Sorry I'm only working of the comment. I'll compare the two and check it out

Comment (AlexanderSehr) [reply avm/res/batch/batch-account/main.bicep:50]: Turns out it's actually a bit more different that I thought. May I suggest that we push the change to the module's logic back?

Comment (eriqua) [reply avm/res/batch/batch-account/main.bicep:50]: OK, but I'll raise one PR to fix this and bump to 0.2.0 before getting this ORPHANED if that is OK

File: avm/utilities/tools/Set-AVMModule.ps1
Comment (eriqua) [inline avm/utilities/tools/Set-AVMModule.ps1:119]: Please resolve conflicts, probably due to readme updates

File: avm/res/batch/batch-account/tests/e2e/encr/main.test.bicep
Comment (eriqua) [inline avm/res/batch/batch-account/tests/e2e/encr/main.test.bicep:15]: It looks like we have Azure Batch accounts per region per subscription default limit set to 3. Should we change the deployment location for one of the tests?

Ref: https://learn.microsoft.com/en-us/azure/batch/batch-quota-limit#resource-quotas

File: avm/res/batch/batch-account/main.bicep
Comment (eriqua) [reply avm/res/batch/batch-account/main.bicep:50]: Done in PR #592

File: avm/res/batch/batch-account/tests/e2e/encr/main.test.bicep
Comment (AlexanderSehr) [reply avm/res/batch/batch-account/tests/e2e/encr/main.test.bicep:15]: Instead of implement a workaround for one resource alone I'm wondering if we should invest some time to implement a CI-powered location rotation. What do you thing @eriqua?

File: avm/utilities/tools/Set-AVMModule.ps1
Comment (AlexanderSehr) [reply avm/utilities/tools/Set-AVMModule.ps1]: done

File: avm/res/batch/batch-account/tests/e2e/encr/main.test.bicep
Comment (AlexanderSehr) [reply avm/res/batch/batch-account/tests/e2e/encr/main.test.bicep]: Updated & retested @eriqua

**Review by eriqua — APPROVED** [review]
:ok:

**Pull request merged**
